enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+type NodeBroadcastOptions struct {
+    // nodes specifies options related to the address that is broadcasted for communication with other nodes.
+    // This field controls the `broadcast_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"ServiceClusterIP"}
+    Nodes *BroadcastOptions `json:"nodes"`
+    
+    // clients specifies options related to the address that is broadcasted for communication with clients.
+    // This field controls the `broadcast_rpc_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"ServiceClusterIP"}
+    Clients *BroadcastOptions `json:"clients"`
+}
+
+type ListenAddressType string
+
+const (
+    // AnyListenAddressType specifies that listen socket will be bound to all local interfaces.
+    AnyListenAddressType ListenAddressType = "Any"
+    
+    // PodIPListenAddressType specifies that listen socket will be bound to Pod IP address.
+    PodIPListenAddressType ListenAddressType = "PodIP"
+)
+
+// ListenOptions hold options related to listening socket of ScyllaDB node.
+type ListenOptions struct {
+    // type of listen address.
+    // +kubebuilder:default:="Any"
+    Type ListenAddressType `json:"type"`
+    
+    // podIP hold options related to PodIP address.
+    // +optional
+    PodIP *PodIPAddressOptions `json:"podIP"`
+}
+
+// NodeListenOptions hold options related to listening on interfaces.
+type NodeListenOptions struct {
+    // nodes specifies options related to the address that is bound to socket used for communication with other nodes.
+    // This field controls the `listen_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"Any"}
+    Nodes ListenOptions `json:"nodes"`
+    
+    // clients specifies options related to the address that is bound to socket used for communication with clients.
+    // This field controls the `rpc_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"Any"}
+    Clients ListenOptions `json:"clients"`
+}
+
+type NodeServiceType string
+
+const (
+    // HeadlessServiceType means nodes will be exposed via Headless Service.
+    HeadlessServiceType NodeServiceType = "Headless"
+    
+    // ClusterIPServiceType means nodes will be exposed via ClusterIP Service.
+    ClusterIPServiceType NodeServiceType = "ClusterIP"
+    
+    // LoadBalancerServiceType means nodes will be exposed via LoadBalancer Service.
+    LoadBalancerServiceType NodeServiceType = "LoadBalancer"
+)
+
+type NodeServiceTemplate struct {
+    // annotations is a custom key value map merged with every node Service annotations.
+    // +optional
+    Annotations map[string]string `json:"annotations,omitempty"`
+    
+    // type is the Kubernetes Service type.
+    // +kubebuilder:validation:Required
+    Type NodeServiceType `json:"type"`
+    
+    // externalTrafficPolicy controls value of corev1/Service.spec.externalTrafficPolicy of each node Service.
```

Comment:
`corev1` isn't ambiguous compared to just `v1`, leaving as is.

Reply:
> corev1 isn't ambiguous compared to just v1, leaving as is.

I said it's not the group name in the API, not that it's ambiguous.
```
$ kubectl get --raw /apis | jq '.groups[].name' | grep -i '"core"'
# <no output>
```
or
```
$ kubectl api-resources | grep -E -e ' APIVERSION | KIND$| Service$'
NAME                              SHORTNAMES   APIVERSION                             NAMESPACED   KIND
services                          svc          v1                                     true         Service
```
(where the APIVERSION field holds `<group>/<version>` for those that have it)

When referencing a field, it would be great if the reference would be passable to `kubectl explain` where users can learn more about it.

```
$ kubectl explain corev1/Service.spec.externalTrafficPolicy
the server doesn't have a resource type "corev1/Service"
```
vs
```
$ kubectl explain service.spec.externalTrafficPolicy
KIND:     Service
VERSION:  v1

FIELD:    externalTrafficPolicy <string>

DESCRIPTION:
...
```